lib/mix/tasks/compiler/gleam.ex

```diff
@@ -1,10 +1,26 @@
 defmodule Mix.Tasks.Compile.Gleam do
   use Mix.Task.Compiler
 
+  @first_gleam_version_not_taking_build_arg "0.18.0-rc1"
+
   def run(_args) do
-    case Mix.shell().cmd("gleam build .") do
+    case Mix.shell().cmd(build_command()) do
       0 -> {:ok, []}
       status -> exit(status)
     end
   end
+
+  defp build_command do
+    if build_needs_arg?(),
+      do: "gleam build .",
```

Comment:
The `.` was optional before, so it could be omitted always, saving the version check

Reply:
ok @lpil, I just simplified the PR based on this insight.  Thanks!